drivers/nrf9160_gps/nrf9160_gps.c

```diff
@@ -96,6 +96,26 @@ static bool is_fix(struct gps_pvt *pvt)
 		== NRF_GNSS_PVT_FLAG_FIX_VALID_BIT);
 }
 
+/**@brief Checks if GPS operation is blocked due to insufficient time windows */
+static bool gps_is_blocked(nrf_gnss_pvt_data_frame_t *pvt)
+{
+/* TODO: Use bitmask from nrf_socket.h when it's available. */
+#define PVT_FLAG_OP_BLOCKED_TIME_WINDOW	BIT(4)
+
+	return ((pvt->flags & PVT_FLAG_OP_BLOCKED_TIME_WINDOW)
+		== PVT_FLAG_OP_BLOCKED_TIME_WINDOW);
+}
+
+/**@brief Checks if PVT frame is invalid and should be discarded */
+static bool pvt_is_invalid(nrf_gnss_pvt_data_frame_t *pvt)
+{
+/* TODO: Use bitmask from nrf_socket.h when it's available. */
+#define PVT_FLAG_CONTENT_INVALID BIT(3)
+
+	return ((pvt->flags & PVT_FLAG_CONTENT_INVALID)
```

Comment:
This is not the invalid flag, it's just why the fix is invalid. Same goes for bit 4, it the reason for why the fix is invalid. You only need to look at the state of bit 0 to see if the fix is valid or not.

And then you can maybe add a mechanism to report to the user why it is invalid based on bit 3 and 4.

Reply:
I've updated the name of the define to `PVT_FLAG_DEADLINE_MISSED` and the function name to `pvt_deadline_missed`. Does it sound okay?

Reasoning for doing the checks is to not propagate events to application if PVT frame is empty (case with blocked operation), or entire content is invalid (case with missed deadline). Even though bit 0 is 0 (and no other bit is set), meaning invalid PVT estimate, it's still useful to the application to be notified with the satellite stats data.